packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -355,35 +356,35 @@ class AndroidDevice extends Device {
       printTrace('Waiting for observatory port to be available...');
 
       try {
-        int observatoryDevicePort, diagnosticDevicePort;
```

Comment:
Agreed but out of scope for this PR. Added a TODO and I'll try and tackle that soon.